src/scripts/split_sssom_by_source.py

```diff
@@ -70,24 +71,23 @@ def read_metadata(filename):
     return meta, curie_map
 
 meta, curie_map=read_metadata(metadata_file)
-df=read_pandas(sssom_file)
-df["mapping_justification"]="semapv:ManualMappingCuration"
-
+msdf_main=parse_sssom_table(sssom_file)
+msdf_main.df["mapping_justification"]=SEMAPV.ManualMappingCuration.value
 subject_prefixes_allowed = meta['subject_prefixes']
 relations_allowed = meta['relations']
 
-subject_prefixes=set(df['subject_id'].str.split(pat=':', n=1, expand=True)[0])
-object_prefixes=set(df['object_id'].str.split(pat=':',n= 1, expand=True)[0])
-relations=set(df['predicate_id'])
 
-df = replace_temporary_prefixes(subject_prefixes, df)
-df = replace_temporary_prefixes(object_prefixes, df)
-global_metadata=meta['global_metadata']
+subject_prefixes=set(msdf_main.df['subject_id'].str.split(pat=':', n=1, expand=True)[0])
+object_prefixes=set(msdf_main.df['object_id'].str.split(pat=':',n= 1, expand=True)[0])
+relations=set(msdf_main.df['predicate_id'])
 
-msdf = from_sssom_dataframe(df, prefix_map=curie_map, meta=global_metadata)
+msdf_main.df = replace_temporary_prefixes(subject_prefixes, msdf_main.df)
```

Comment:
I won't open up the repo to examine how this func works but I can guess the idea.